### ISSUE TITLE
Crash report from Crashlytics

### ISSUE THREAD

**Issue by mnizar** [jwplayer/JWPlayer-iOS-SDK-cocoapod#17]
I'm using v2.8.2

`Fatal Exception: NSInvalidArgumentException
*** -[_NSPlaceholderData initWithBase64EncodedString:options:]: nil string argument`

```
Fatal Exception: NSInvalidArgumentException
0  (Missing)                      0x18d556fe0 (Missing)
1  (Missing)                      0x18bfb8538 (Missing)
2  (Missing)                      0x18d556f28 (Missing)
3  (Missing)                      0x18dfbe56c (Missing)
4  JWPlayer_iOS_SDK               0x101082658 (Missing)
5  (Missing)                      0x18d5491c0 (Missing)
6  JWPlayer_iOS_SDK               0x101082510 (Missing)
7  JWPlayer_iOS_SDK               0x1010814c8 (Missing)
8  (Missing)                      0x18c40e9e0 (Missing)
9  (Missing)                      0x18c40e9a0 (Missing)
10 (Missing)                      0x18c41d164 (Missing)
11 (Missing)                      0x18c41ea50 (Missing)
12 (Missing)                      0x18c41e7d0 (Missing)
13 (Missing)                      0x18c617100 (Missing)
14 (Missing)                      0x18c616cac (Missing)
```

`JWPlayer_iOS_SDK
__36-[JWFileDownloader createLocalFiles]_block_invoke`

```Crashed: com.twitter.crashlytics.ios.exception
0  MyApp                      0x1003439c8 CLSProcessRecordAllThreads + 4299078088
1  MyApp                      0x1003439c8 CLSProcessRecordAllThreads + 4299078088
2  MyApp                      0x100343884 CLSProcessRecordAllThreads + 4299077764
3  MyApp                      0x1003335c4 CLSHandler + 4299011524
4  MyApp                      0x1003419d0 __CLSExceptionRecord_block_invoke + 4299069904
5  (Missing)                      0x18c40e9a0 (Missing)
6  (Missing)                      0x18c41bee0 (Missing)
7  MyApp                      0x100341464 CLSExceptionRecord + 4299068516
8  MyApp                      0x10034128c CLSExceptionRecordNSException + 4299068044
9  MyApp                      0x100340ea0 CLSTerminateHandler() + 4299067040
10 (Missing)                      0x18bfa75d4 (Missing)
11 (Missing)                      0x18bfa6ef8 (Missing)
12 (Missing)                      0x18bfb866c (Missing)
13 (Missing)                      0x18d556f28 (Missing)
14 (Missing)                      0x18dfbe56c (Missing)
15 JWPlayer_iOS_SDK               0x101082658 __36-[JWFileDownloader createLocalFiles]_block_invoke + 256
16 (Missing)                      0x18d5491c0 (Missing)
17 JWPlayer_iOS_SDK               0x101082510 -[JWFileDownloader createLocalFiles] + 168
18 JWPlayer_iOS_SDK               0x1010814c8 __45-[JWFileDownloader fetchFilesWithCompletion:]_block_invoke + 384
19 (Missing)                      0x18c40e9e0 (Missing)
20 (Missing)                      0x18c40e9a0 (Missing)
21 (Missing)                      0x18c41d164 (Missing)
22 (Missing)                      0x18c41ea50 (Missing)
23 (Missing)                      0x18c41e7d0 (Missing)
24 (Missing)                      0x18c617100 (Missing)
25 (Missing)                      0x18c616cac (Missing)

**Comment by okulagin** [comment]
Hi @mnizar,

I do apologize for the delay. We are looking into it and will get back to this thread within a couple of days.

**Comment by rodrigoschmitt** [comment]
Hi @okulagin, do you have more information about this bug? Because We have the same issue in our App.

We're using the v2.8.6+44 

```
[NSData(NSData) initWithBase64EncodedString:options:]
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[_NSPlaceholderData initWithBase64EncodedString:options:]: nil string argument'
1  CoreFoundation  __exceptionPreprocess + 124
2  libobjc.A.dylib  objc_exception_throw + 52
3  CoreFoundation  +[NSException raise:format:] + 112
4  Foundation  -[NSData(NSData) initWithBase64EncodedString:options:] + 96
5  JWPlayer_iOS_SDK  0x105608000 + 288624
6  CoreFoundation  -[__NSArrayM enumerateObjectsWithOptions:usingBlock:] + 212
7  JWPlayer_iOS_SDK  0x105608000 + 288296
8  JWPlayer_iOS_SDK  0x105608000 + 284128
9  libdispatch.dylib  _dispatch_call_block_and_release + 20
```

**Comment by okulagin** [comment]
@rodrigoschmitt Unfortunately not yet. We got some urgent tasks to do and we've been busy with them. But anyway it is on our radar.

**Comment by okulagin** [comment]
@rodrigoschmitt I'm just wondering whether you have steps to reproduce. Thanks

**Comment by mnizar** [comment]
Hi @okulagin ,
we never got the crash, but the crashlytics says its crash and its number one and two on list crashes crashlytics 
![13-04-2018-12-53-20](https://user-images.githubusercontent.com/4274589/38718959-c538eafc-3f19-11e8-87de-3f8c8a93f5a0.JPG)

**Comment by okulagin** [comment]
@mnizar it is supposed to be fixed in 2.8.10

**Comment by mnizar** [comment]
Yes, it fixed, never appear again in crashlytics
thank you @okulagin
